include/jwt-cpp/traits/nlohmann-json/traits.h

```diff
@@ -64,7 +64,8 @@ namespace jwt {
 				return val.get<double>();
 			}
 
-			static bool parse(json& val, std::string str) {
+			template<class string_t>
+			static bool parse(json& val, const string_t& str) {
```

Comment:
Would it not be more ideal to specialize these so users do not make mistakes

https://github.com/nlohmann/json/blob/6af826d0bdb55e4b69e3ad817576745335f243ca/README.md?plain=1#L1640

There could be one for both types?